extension/encoding/awscloudwatchmetricstreamsencodingextension/extension_test.go

```diff
@@ -26,7 +28,9 @@ func TestNew_OpenTelemetry10(t *testing.T) {
 	require.NotNil(t, e)
 
 	_, err = e.UnmarshalMetrics([]byte{})
-	assert.EqualError(t, err, `UnmarshalMetrics unimplemented for format "opentelemetry1.0"`)
+	expectedErr := formatOpenTelemetryError(errEmptyRecord)
+	require.Error(t, expectedErr)
+	assert.EqualError(t, err, expectedErr.Error())
```

Comment:
```suggestion
	assert.EqualError(t, err, `failed to unmarshal metrics as 'opentelemetry1.0' format: 0 metrics were extracted from the record`)
```

(Yes I realise this isn't very DRY, but that's less important than ensuring the error messages are & remain meaningful over time.)